### ISSUE TITLE
Separate VFS used specifically for user data

### ISSUE THREAD

**Issue by alula** [ggez/ggez#881]
The current filesystem API design combines both user and data directories into one virtual filesystem, which means that user files might unintentionally overwrite game resources, or certain resources can be treated as user files.

I suggest removing/deprecating/keeping the current filesystem methods (idk what people would prefer) and providing methods that return a reference to user/resources filesystems:
```rust
// before
let mut file = filesystem::open(ctx, "/data/tile.map")?;
file.read_to_end(&mut buffer)?;

let mut file = filesystem::create(ctx, "save.dat")?;
file.write_all(bytes)?;

// after
let mut file = filesystem::resources(ctx).open("/data/tile.map")?;
file.read_to_end(&mut buffer)?;

let mut file = filesystem::user(ctx).create("save.dat")?;
file.write_all(bytes)?;
```

**Comment by icefoxen** [comment]
It merges user and data directories into one virtual file system, with the user directories being writeable and the data directories being read-only.

**Comment by alula** [comment]
whoops I meant shadowing

**Comment by alula** [comment]
Also after messing around with code I've noticed that simply exposing VFS structs/methods in ggez API and adding ability to skip data directory mount could be one of solutions to described problem - no API breakage would be needed.

**Comment by icefoxen** [comment]
Yeah that's been discussed in other issues: #873 and #767 .  Splitting out the ggez `vfs` module into its own crate is something that's been desirable for a long time as well.  There's an experimental implementation of it here: https://github.com/ggez/ggez/tree/ggraphics/gvfs but it will need some work to nicely work into ggez itself.  Contributions welcome!  :D
